runtime: `cargo test` fails with modern GCC

`cargo test` fails to build with newer gcc (11.1.1) due to C dependencies pulled in by our ancient version of `grpcio-sys`.

```
  /home/yawning/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.4.7/grpc/third_party/boringssl/third_party/fiat/curve25519.c:1905:57: error: argument 2 of type ‘const uint8_t *’ {aka ‘const unsigned char *’} declared as a pointer [-Werror=array-parameter=]
   1905 | void x25519_ge_scalarmult_base(ge_p3 *h, const uint8_t *a) {
        |                                          ~~~~~~~~~~~~~~~^
  In file included from /home/yawning/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.4.7/grpc/third_party/boringssl/third_party/fiat/curve25519.c:41:
  /home/yawning/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.4.7/grpc/third_party/boringssl/third_party/fiat/internal.h:125:56: note: previously declared as an array ‘const uint8_t[32]’ {aka ‘const unsigned char[32]’}
    125 | void x25519_ge_scalarmult_base(ge_p3 *h, const uint8_t a[32]);
        |                                          ~~~~~~~~~~~~~~^~~~~
  cc1: all warnings being treated as errors
  gmake[3]: *** [third_party/boringssl/third_party/fiat/CMakeFiles/fiat.dir/build.make:82: third_party/boringssl/third_party/fiat/CMakeFiles/fiat.dir/curve25519.c.o] Error 1
  gmake[2]: *** [CMakeFiles/Makefile2:3544: third_party/boringssl/third_party/fiat/CMakeFiles/fiat.dir/all] Error 2
  gmake[2]: *** Waiting for unfinished jobs....
  gmake[1]: *** [CMakeFiles/Makefile2:1727: CMakeFiles/grpc.dir/rule] Error 2
  gmake: *** [Makefile:300: grpc] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2
```

> Can you check if it works with #3950?

```
  /home/yawning/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.8.1/grpc/third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc: In member function ‘void absl::lts_2020_09_23::synchronization_internal::GraphCycles::RemoveNode(void*)’:
  /home/yawning/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.8.1/grpc/third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc:451:26: error: ‘numeric_limits’ is not a member of ‘std’
    451 |   if (x->version == std::numeric_limits<uint32_t>::max()) {
        |                          ^~~~~~~~~~~~~~
  /home/yawning/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.8.1/grpc/third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc:451:49: error: expected primary-expression before ‘>’ token
    451 |   if (x->version == std::numeric_limits<uint32_t>::max()) {
        |                                                 ^
  /home/yawning/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.8.1/grpc/third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc:451:52: error: ‘::max’ has not been declared; did you mean ‘std::max’?
    451 |   if (x->version == std::numeric_limits<uint32_t>::max()) {
        |                                                    ^~~
        |                                                    std::max
  In file included from /usr/include/c++/11/algorithm:62,
                   from /home/yawning/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.8.1/grpc/third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc:38:
  /usr/include/c++/11/bits/stl_algo.h:3467:5: note: ‘std::max’ declared here
   3467 |     max(initializer_list<_Tp> __l, _Compare __comp)
        |     ^~~
  gmake[3]: *** [third_party/abseil-cpp/absl/synchronization/CMakeFiles/absl_graphcycles_internal.dir/build.make:82: third_party/abseil-cpp/absl/synchronization/CMakeFiles/absl_graphcycles_internal.dir/internal/graphcycles.cc.o] Error 1
  gmake[2]: *** [CMakeFiles/Makefile2:3340: third_party/abseil-cpp/absl/synchronization/CMakeFiles/absl_graphcycles_internal.dir/all] Error 2
  gmake[2]: *** Waiting for unfinished jobs....
  gmake[1]: *** [CMakeFiles/Makefile2:1009: CMakeFiles/grpc.dir/rule] Error 2
  gmake: *** [Makefile:261: grpc] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2
```

It managed to move the error around at least.

--- a/.changelog/3968.internal.md
+++ b/.changelog/3968.internal.md
@@ -1,0 +1,1 @@
+Bump grpcio to 0.9.0

--- a/client/Cargo.toml
+++ b/client/Cargo.toml
@@ -16,7 +16,7 @@ tokio = { version = "1", features = ["rt", "sync"] }
 io-context = "0.2.0"
 
 [target.'cfg(not(target_env = "sgx"))'.dependencies]
-grpcio = "0.8.2"
+grpcio = "0.9.0"
 rustracing = "0.5.1"
 rustracing_jaeger = "0.6.1"
 tokio-stream = { version = "0.1.6", features = ["sync"] }

--- a/client/src/transaction/client.rs
+++ b/client/src/transaction/client.rs
@@ -2,7 +2,7 @@
 use std::time::Duration;
 
 use futures::prelude::*;
-use grpcio::{Channel, Error::RpcFailure, RpcStatus, RpcStatusCode};
+use grpcio::{Channel, Error::RpcFailure, RpcStatusCode};
 use rustracing::{sampler::AllSampler, tag};
 use rustracing_jaeger::{span::Span, Tracer};
 use serde::{de::DeserializeOwned, Serialize};
@@ -170,10 +170,7 @@ impl TxnClient {
             .await;
 
         match rsp {
-            Err(RpcFailure(RpcStatus {
-                status: RpcStatusCode::NOT_FOUND,
-                ..
-            })) => Ok(None),
+            Err(RpcFailure(s)) if s.code() == RpcStatusCode::NOT_FOUND => Ok(None),
             Err(err) => Err(TxnClientError::CallFailed(format!("{}", err))),
             Ok(rsp) => Ok(Some(BlockSnapshot::new(storage_client, rsp))),
         }
@@ -200,10 +197,7 @@ impl TxnClient {
             .await;
 
         match rsp {
-            Err(RpcFailure(RpcStatus {
-                status: RpcStatusCode::NOT_FOUND,
-                ..
-            })) => Ok(None),
+            Err(RpcFailure(s)) if s.code() == RpcStatusCode::NOT_FOUND => Ok(None),
             Err(err) => Err(TxnClientError::CallFailed(format!("{}", err))),
             Ok(rsp) => Ok(Some(
                 TransactionSnapshot::new(storage_client, rsp.block, index, rsp.input, rsp.output)
@@ -233,10 +227,7 @@ impl TxnClient {
             .await;
 
         match rsp {
-            Err(RpcFailure(RpcStatus {
-                status: RpcStatusCode::NOT_FOUND,
-                ..
-            })) => Ok(None),
+            Err(RpcFailure(s)) if s.code() == RpcStatusCode::NOT_FOUND => Ok(None),
             Err(err) => Err(TxnClientError::CallFailed(format!("{}", err))),
             Ok(rsp) => Ok(Some(
                 TransactionSnapshot::new(storage_client, rsp.block, index, rsp.input, rsp.output)
@@ -281,10 +272,7 @@ impl TxnClient {
             .await;
 
         match rsp {
-            Err(RpcFailure(RpcStatus {
-                status: RpcStatusCode::NOT_FOUND,
-                ..
-            })) => Ok(None),
+            Err(RpcFailure(s)) if s.code() == RpcStatusCode::NOT_FOUND => Ok(None),
             Err(err) => Err(TxnClientError::CallFailed(format!("{}", err))),
             Ok(rsp) => Ok(Some(BlockSnapshot::new(storage_client, rsp))),
         }
@@ -315,10 +303,7 @@ impl TxnClient {
             .await;
 
         match rsp {
-            Err(RpcFailure(RpcStatus {
-                status: RpcStatusCode::NOT_FOUND,
-                ..
-            })) => Ok(None),
+            Err(RpcFailure(s)) if s.code() == RpcStatusCode::NOT_FOUND => Ok(None),
             Err(err) => Err(TxnClientError::CallFailed(format!("{}", err))),
             Ok(rsp) => Ok(Some(
                 TransactionSnapshot::new(

--- a/keymanager-client/Cargo.toml
+++ b/keymanager-client/Cargo.toml
@@ -14,4 +14,4 @@ lru = "0.6.5"
 thiserror = "1.0"
 
 [target.'cfg(not(target_env = "sgx"))'.dependencies]
-grpcio = "0.8.2"
+grpcio = "0.9.0"

--- a/runtime/Cargo.toml
+++ b/runtime/Cargo.toml
@@ -55,7 +55,7 @@ impl-trait-for-tuples = "0.2.1"
 
 [dev-dependencies]
 # For storage interoperability tests only.
-grpcio = "0.8.2"
+grpcio = "0.9.0"
 tempfile = "3.2.0"
 
 [[bin]]

--- a/runtime/Cargo.toml
+++ b/runtime/Cargo.toml
@@ -39,7 +39,7 @@ io-context = "0.2.0"
 curve25519-dalek = "3.0.0"
 x25519-dalek = "1.1.0"
 ed25519-dalek = "1.0.1"
-deoxysii = { git = "https://github.com/oasisprotocol/deoxysii-rust" }
+deoxysii = "0.2.2"
 tiny-keccak = { version = "2.0.2", features = ["sha3"] }
 sp800-185 = "0.2.0"
 zeroize = "1.3"

--- a/tests/clients/simple-keyvalue-enc/Cargo.toml
+++ b/tests/clients/simple-keyvalue-enc/Cargo.toml
@@ -12,7 +12,7 @@ oasis-core-runtime = { path = "../../../runtime" }
 simple-keyvalue-api = { path = "../../runtimes/simple-keyvalue/api" }
 
 clap = "2.29.1"
-grpcio = "0.8.2"
+grpcio = "0.9.0"
 io-context = "0.2.0"
 rand = "0.7.3"
 tokio = { version = "1", features = ["rt-multi-thread"] }

--- a/tests/clients/simple-keyvalue-ops/Cargo.toml
+++ b/tests/clients/simple-keyvalue-ops/Cargo.toml
@@ -11,7 +11,7 @@ oasis-core-runtime = { path = "../../../runtime" }
 simple-keyvalue-api = { path = "../../runtimes/simple-keyvalue/api" }
 
 clap = "2.29.1"
-grpcio = "0.8.2"
+grpcio = "0.9.0"
 io-context = "0.2.0"
 rand = "0.7.3"
 tokio = { version = "1", features = ["rt-multi-thread"] }

--- a/tests/clients/simple-keyvalue/Cargo.toml
+++ b/tests/clients/simple-keyvalue/Cargo.toml
@@ -11,7 +11,7 @@ oasis-core-runtime = { path = "../../../runtime" }
 simple-keyvalue-api = { path = "../../runtimes/simple-keyvalue/api" }
 
 clap = "2.29.1"
-grpcio = "0.8.2"
+grpcio = "0.9.0"
 io-context = "0.2.0"
 rand = "0.7.3"
 tokio = { version = "1", features = ["rt-multi-thread"] }

--- a/tests/clients/test-long-term/Cargo.toml
+++ b/tests/clients/test-long-term/Cargo.toml
@@ -11,6 +11,6 @@ oasis-core-runtime = { path = "../../../runtime" }
 simple-keyvalue-api = { path = "../../runtimes/simple-keyvalue/api" }
 
 clap = "2.29.1"
-grpcio = "0.8.2"
+grpcio = "0.9.0"
 rand = "0.7.3"
 tokio = { version = "1", features = ["rt-multi-thread"] }

